src/lib/containers/markdown/MarkdownEditor.tsx

```diff
@@ -17,7 +22,62 @@ export const MarkdownEditor: React.FunctionComponent<MarkdownEditorProps> = ({
   const [currentTab, setCurrentTab] = useState<MarkdownEditorTabs>(
     MarkdownEditorTabs.WRITE,
   )
-  const [text, setText] = useState<string>()
+  const [text, setText] = useState<string>('')
+  const [selectionStart, setSelectionStart] = useState<number>(0)
+  const textAreaRef = useRef<HTMLTextAreaElement>(null)
+
+  useEffect(() => {
```

Comment:
```suggestion
  /**
   * Updates the position of the cursor. This must occur within `useEffect` because we need to ensure that the
   * cursor position is updated after the state of `text` is updated.
   */
  useEffect(() => {
```